src/nvim/charset.c

```diff
@@ -1740,11 +1760,8 @@ char_u* skiptowhite_esc(char_u *p) {
 /// @return Number read from the string.
 intmax_t getdigits(char_u **pp)
 {
-  errno = 0;
-  intmax_t number = strtoimax((char *)*pp, (char **)pp, 10);
-  if (number == INTMAX_MAX || number == INTMAX_MIN) {
-    assert(errno != ERANGE);
-  }
+  intmax_t number;
+  assert(getdigits_safe(pp, &number) == 0);
```

Comment:
Good catch, thanks.

I'm currently traveling and I'll address all the comments once I'm back.